docs: add angular with vite example

drdreo opened the pull request:
I've created a plain Angular project using their CLI. With ng version 17.3 it also comes with vite so it could fit into that category.
The project also demos quickly how to use the plugins to get started.

### Intention
I was not finding a single reference regarding Angular in the UnoCSS documentation which lead to the conclusion Angular is not supported. This is very much the opposite and I want to help to change that perception. 

It would have great benefits having at least a demo example referenced inside the official docs. This could spread awareness of this great tool amongst the ng-folk.

zyyv commented:
That's a great start, and you can guide the other example, remove some unnecessary files, make it minimal.

drdreo commented:
@zyyv thanks for the feedback. I have no removed all obsolete files, to have a minimal, servable project and also added the common stackblitz file (assuming this is needed for demo purposes).
I also added the entry to the examples cards here: [https://unocss.dev/integrations/#examples](https://unocss.dev/integrations/#examples)

Anything else you want me to add?

userquin commented on examples/vite-angular/package.json:
```diff
@@ -0,0 +1,41 @@
+    "karma-jasmine-html-reporter": "~2.1.0",
+    "unocss": "^0.60.0",
+    "typescript": "~5.4.2"
+  }
```
Add this entry:
```json
  "stackblitz": {
    "installDependencies": false,
    "startCommand": "node .stackblitz.mjs && npm install && npm run dev"
  }
```

userquin commented on examples/vite-angular/.stackblitz.mjs:
```diff
@@ -0,0 +1,14 @@
+function updateDependencyLinksToLatest(filename) {
+  try {
+    const contents = fs.readFileSync(filename, 'utf-8')
+    const updatedContent = contents.replace(/"link:...+"/gi, '"latest"')
```
can you change the regex to: `/"link:.{3,}"/gi`?

userquin replied:
for context: https://github.com/unocss/unocss/pull/3801

userquin commented on examples/vite-angular/package.json:
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "ng-uno",
+  "version": "0.0.0",
```
shouldn't add `"type": "module"`?

userquin replied:
UnoCSS is ESM only...

userquin commented:
> I also added the entry to the examples cards here: https://unocss.dev/integrations/#examples But the `i-logos-angular` doesn't seem to exist yet.

https://icones.js.org/collection/logos?s=angular

userquin commented:
@drdreo you can test your branch in SB: https://stackblitz.com/github/drdreo/unocss/tree/docs-vite-angular/examples/vite-angular

userquin commented:
@drdreo can you run in root folder `nr lint` to fix lint errors?

drdreo commented:
@userquin My apologies, i had to setup pnpm first. Now its linted properly and i factored in your feedback.

userquin commented:
I'll check the PR later...

drdreo commented:
Yeah, so before Angular builds, we need the referenced `uno.css` file which is generated by the unocss cli.
For that i have added the `"dev": "npm run build:css && npm run start",` script which works for me locally when ran. This is also used in the stackblitz.mjs but it seems the preview didn't update yet.

userquin commented:
I'm going to include uno.css in `.gitignores` and fix pnpm workspace for the example... I'm also including `cli.analytics = false`  in angular.json.

userquin commented:
No idea what's wrong with my previous VP link, cloned repo can be found here: https://stackblitz.com/edit/stackblitz-starters-gybg14?file=package.json,src%2Funo.css
